neuro_flow/context.py

```diff
@@ -546,14 +546,20 @@ async def setup_defaults_env_tags_ctx(
     ctx: WithFlowContext,
     ast_defaults: Optional[ast.FlowDefaults],
 ) -> Tuple[DefaultsConf, EnvCtx, TagsCtx]:
+    env: EnvCtx
+    tags: TagsCtx
     if ast_defaults is not None:
         if ast_defaults.env is not None:
-            env = {k: await v.eval(ctx) for k, v in ast_defaults.env.items()}
+            tmp_env = await ast_defaults.env.eval(ctx)
+            assert isinstance(tmp_env, dict)
```

Comment:
Minor. To be honest, I feel uncomfortable when I see such asserts. I suggest two points here: 
- We can make create `StrictBaseExpr` to check for `None` in compile type (both `MappingExpr` and `MappingItemsExpr` never produce `None`)
- We can use `dict(tmp_env)` to convert `MappingT` to `dict`. It will do nothing for dicts, but it looks clearer as for me. 
We can probably do the same for `SequenceExpr` and `SequenceItemsExpr`.

Reply:
The proposal doesn't work with mypy :(